### DISCUSSION TITLE
Behaviour of clause 'on conflict' with sqlite

### DISCUSSION THREAD

**Discussion by DSMejantel** [sqlpage/SQLPage#176] [Q&A]
Hello,
I don't understand something.
I'm working with SQLpage and sqlite for my apli 'Ecole Inclusive'
I have do use IGNORE instead of ON CONFLICT ... DO NOTHING
In another case, SQLpage return an error (near "DO" : syntax error)
For example in this file comptes_upload.sql :

```sql
-- insert the data into the final table
INSERT OR IGNORE INTO user_info (username, password_hash, nom, prenom, tel, courriel, groupe)
select username, password, nom, prenom, tel, courriel, CAST(groupe AS integer)   from user_tmp;
```

But, in an another file create_user.sql : 

```sql
INSERT INTO user_info (username, password_hash, nom, prenom, groupe, tel, courriel)
    VALUES (:username, sqlpage.hash_password(:password), :nom, :prenom, :groupe, :tel, :courriel)
    ON CONFLICT (username) DO NOTHING
    RETURNING 
     'redirect' AS component,
    'create_user_welcome_message.sql?username=' || :username AS link;
```
the clause DO NOTHING don't return error and all is ok.
How can we explain this incredible behaviour ?

**Comment by lovasoa** [comment]
This is explained in the SQLite documentation:

 - https://www.sqlite.org/lang_upsert.html
 - https://www.sqlite.org/lang_conflict.html

> The phrase "ON CONFLICT" is also part of [UPSERT](https://www.sqlite.org/lang_upsert.html), which is an extension to [INSERT](https://www.sqlite.org/lang_insert.html) added in version 3.24.0 (2018-06-04). Do not confuse these two separate uses of the "ON CONFLICT" phrase.
